cmake/modules/gd.cmake

```diff
@@ -7,10 +7,10 @@ MACRO(ADD_GD_TESTS)
 	FOREACH(test_name ${TESTS_FILES})
 		SET(test_prog_name "test_${TEST_PREFIX}_${test_name}")
 		add_executable(${test_prog_name} "${test_name}.c")
-		IF(WIN32)
-			target_link_libraries (${test_prog_name} gdTest ${ARGV0})
-		ELSE(WIN32)
+		IF(NEED_LINKING_AGAINST_LIBM)
```

Comment:
we shouldn't need this at all.  libgd.so itself should link against libm.  pretty sure the m linkage here was to workaround us missing it in the main library.

Reply:
Hmm, but what if the user builds a static libgd?

Reply:
Yeah, you're right. I'll work on that soon.

Reply:
Indeed, looks like this is already properly handled by `target_link_libraries()` for libgd and gdTest, respectively.